Assets/LeapMotionModules/Interaction/InteractionHand.cs

```diff
@@ -0,0 +1,800 @@
+      if (hoverResults.primaryHovered == _primaryHoveredLastFrame) {
+        if (hoverResults.primaryHovered != null) hoverResults.primaryHovered.PrimaryHoverStay(hoverResults.checkedHand);
+      }
+      else {
+        if (_primaryHoveredLastFrame != null) {
+          _primaryHoveredLastFrame.PrimaryHoverEnd(hoverResults.checkedHand);
+        }
+        _primaryHoveredLastFrame = hoverResults.primaryHovered;
+        if (_primaryHoveredLastFrame != null) _primaryHoveredLastFrame.PrimaryHoverBegin(hoverResults.checkedHand);
+      }
+    }
+
+    private const float MAX_PRIMARY_HOVER_DISTANCE = 0.1F;
+    private Vector3 _intentionPosition;
+    private InteractionBehaviourBase _primaryHoveredLastFrame = null;
+    private float[] _intentionFingerWeights = new float[NUM_FINGERS] { 0F, 1F, 0.6F, 0.4F, 0.1F };
+
+    // Note. This is used as the hover sphere-check origin, but distance data is not calculated from it.
+    // If it's worthwhile, it should be used to provide distance in hover callbacks... otherwise it should just be cut.
+    private void CalculateIntentionPosition() {
+      if (_hand == null) return;
+
+      // Weighted average of medial finger bone positions
+      float usageSum = 0F;
+      Leap.Vector averagePosition = Leap.Vector.Zero;
+      //Leap.Vector averageDirection = Leap.Vector.Zero;
+      for (int i = 1; i < 5; i++) {
+        Leap.Vector distalFingerBoneTip = _hand.Fingers[i].bones[3].NextJoint;
+        Leap.Vector medialFingerBoneDirection = _hand.Fingers[i].bones[2].Direction;
+        Leap.Vector distalDirection = _hand.Basis.zBasis;
+        var fingerUsage = (((medialFingerBoneDirection.x * distalDirection.x)
+                         + (medialFingerBoneDirection.y * distalDirection.y)
+                         + (medialFingerBoneDirection.z * distalDirection.z))
+                         * _intentionFingerWeights[i])
+                         .Map(-0.3F, 0.9F, 0, 1);
+        usageSum += fingerUsage;
+        averagePosition = new Leap.Vector(averagePosition.x + distalFingerBoneTip.x * fingerUsage,
+                                          averagePosition.y + distalFingerBoneTip.y * fingerUsage,
+                                          averagePosition.z + distalFingerBoneTip.z * fingerUsage);
+        //averageDirection = new Leap.Vector(averageDirection.x + medialFingerBoneDirection.x * fingerUsage,
+        //                                   averageDirection.y + medialFingerBoneDirection.y * fingerUsage,
+        //                                   averageDirection.z + medialFingerBoneDirection.z * fingerUsage);
+      }
+
+      // Distal Intention Ray: Add punch vector for when finger usage is low (closed fist).
+      // Vector3 distalAxis = _hand.DistalAxis();
+      float punchWeight = usageSum.Map(0F, 1F, 1F, 0F);
+      averagePosition = new Leap.Vector(averagePosition.x + _hand.Fingers[2].bones[1].PrevJoint.x * punchWeight,
```

Comment:
Looks like these comments didn't get addressed, any reason you are doing the component math manually instead of just using the vector class?